filter/filter.go

```diff
@@ -10,20 +10,20 @@ import (
 // Filter is a special http.Handler that returns true or false depending on
 // whether subsequent handlers should continue.
 type Filter interface {
-	// ServeHTTP is like the function on http.Handler but also returns true or
-	// false depending on whether subsequent handlers should continue. If an error
-	// occurred, ServeHTTP should return the original error plus a description
-	// for logging purposes.
-	Apply(w http.ResponseWriter, req *http.Request) (ok bool, err error, errdesc string)
+	// ServeHTTP is like the function on http.Handler but also gets a context
+	// which allows it to control the progress of the filter chain.
+	Apply(w http.ResponseWriter, req *http.Request, ctx Context)
```

Comment:
I find the concept of context confusing here. I can see that is a Filter.Context, but we also have requests context and then there is those errors with context... I suggest that we call it something else, like `Next`; Next.Continue(), Next.Fail()... or something different but differentiated from Context.